st2common/st2common/rbac/resolvers.py

```diff
@@ -1047,6 +1048,76 @@ def user_has_permission(self, user_db, permission_type):
         return self._user_has_global_permission(user_db=user_db, permission_type=permission_type)
 
 
+class InquiryPermissionsResolver(PermissionsResolver):
+    resource_type = ResourceType.INQUIRY
+    view_grant_permission_types = [
+        PermissionType.INQUIRY_LIST,
+        PermissionType.INQUIRY_VIEW,
+        PermissionType.INQUIRY_RESPOND,
+        PermissionType.INQUIRY_ALL
+    ]
+
+    def user_has_permission(self, user_db, permission_type):
+        assert permission_type in [PermissionType.INQUIRY_LIST, PermissionType.INQUIRY_ALL]
+        return self._user_has_list_permission(user_db=user_db, permission_type=permission_type)
+
+    def user_has_resource_db_permission(self, user_db, resource_db, permission_type):
+        """
+        Method for checking user permissions on an existing resource (e.g. get one, edit, delete
+        operations).
+
+        NOTE:
+        Because we're borrowing the ActionExecutionDB model, the resource_db parameter is
+        effectively ignored. All other filters are passed to get_all_permission_grants_for_user,
+        then a custom filtering method is performed on the result to enforce the required
+        "inquiry:ask" resource UID passed by the user in the role. This isn't ideal, as the whole
+        premise of a resource_db permission is that it is enforced on an actual DB resource,
+        but it works for now, and it sets the tone for how this will be done when/if Inquiries
+        get their own data model.
+        """
+
+        assert permission_type in [
+            PermissionType.INQUIRY_RESPOND,
+            PermissionType.INQUIRY_VIEW,
+            PermissionType.INQUIRY_ALL
+        ]
+
+        log_context = {
+            'user_db': user_db,
+            'resource_db': resource_db,
+            'permission_type': permission_type,
+            'resolver': self.__class__.__name__
+        }
+        self._log('Checking user resource permissions', extra=log_context)
+
+        # First check the system role permissions
+        has_system_role_permission = self._user_has_system_role_permission(
+            user_db=user_db, permission_type=permission_type)
+
+        if has_system_role_permission:
+            self._log('Found a matching grant via system role', extra=log_context)
+            return True
+
+        resource_types = [ResourceType.INQUIRY]
+        permission_types = [
+            PermissionType.INQUIRY_VIEW,
+            PermissionType.INQUIRY_RESPOND,
+            PermissionType.INQUIRY_ALL
+        ]
+        permission_grants = get_all_permission_grants_for_user(user_db=user_db,
+                                                               resource_types=resource_types,
```

Comment:
Since no resource_id is passed here it looks like those permissions should be global (don't apply to a specific resource) if that's not already the case.

Reply:
Actually, I just saw the code below. I see the approach you went with.

Another approach would be to make those permission types global (don't apply to a specific resource), then you could get rid of that if len check below.

Reply:
This was mostly because Inquiries are still just executions under the hood so I wasn't sure of a better way. I'll see if I can convert the permissions to a global type and get rid of this hack.

Reply:
I took a stab at this in 7d828a2f5. It seems to work (at least the tests are happy, will do some additional testing later) and I'm very happy to `s/inquiry:ask/inquiry` everywhere but I'm not confident that I satisfied your statement: `you could get rid of that if len check below`. For now, I'm assuming by that you meant getting rid of the list comprehension, since there are other resolvers that do a `len()` check, just in a much simpler way (just checking total len of `permission_grants`). Let me know if I missed something there.